aiida/tools/importexport/archive/readers.py

```diff
@@ -357,26 +369,19 @@ def iter_node_repos(self,
         # unarchive the common folder if it does not exist
         common_prefix = os.path.commonprefix(path_prefixes)
         if not self._sandbox.get_subfolder(common_prefix).exists():
-            self._extract(path_prefix=common_prefix, progress=progress)
-
-        if progress:
-            with get_progress_reporter()(total=len(path_prefixes), desc=description) as report:
-                for uuid, path_prefix in zip(uuids, path_prefixes):
-                    report.update()
-                    subfolder = self._sandbox.get_subfolder(path_prefix)
-                    if not subfolder.exists():
-                        raise CorruptArchive(
-                            f'Unable to find the repository folder for Node with UUID={uuid} in the archive'
-                        )
-                    yield subfolder
-        else:
-            for uuid, path_prefix in zip(uuids, path_prefixes):
-                subfolder = self._sandbox.get_subfolder(path_prefix)
-                if not subfolder.exists():
-                    raise CorruptArchive(
-                        f'Unable to find the repository folder for Node with UUID={uuid} in the exported file'
-                    )
-                yield subfolder
+            self._extract(path_prefix=common_prefix, callback=callback)
+
+        if callback is not None:
```

Comment:
Or one could default to a dummy callback, which would make using this feel more like using a logger.

Reply:
I've add a `null_callback` function as default. Is this what you had in mind?

Reply:
exactly, thanks!